lib/intercom-rails/encrypted_mode.rb

```diff
@@ -0,0 +1,34 @@
+module IntercomRails
+  class EncryptedMode
+    attr_reader :secret, :initialization_vector, :enabled
+
+    ENCRYPTED_MODE_SETTINGS_WHITELIST = [:app_id, :session_duration, :widget, :custom_launcher_selector, :hide_default_launcher, :alignment, :horizontal_padding, :vertical_padding]
+
+    def initialize(secret, initialization_vector, options)
```

Comment:
Note that it is optional, we fall back to SecureRandom https://github.com/intercom/intercom-rails/pull/263/files#diff-a86f56ee7fcbd0a1aac0f6800d8c0435R9